src/interpolation.jl

```diff
@@ -10,38 +10,58 @@ time ``x(t)``; or a lunar core Euler angle as a function of time ``\theta_c(t)``
 # Fields
 
 - `t0::T`: Start time.
-- `t::AbstractVector{T}`: Vector of time instances when the timespan of the ``i``-th element of `x` ends and the ``(i+1)``-th element of `x` starts being valid. 
-- `x::AbstractArray{Taylor1{U},N}`: Vector of Taylor polynomials that interpolate the dependent variable as a function of the independent variable.
+- `t::Vector{T}`: Vector of time instances when the timespan of the ``i``-th element of `x` ends and the ``(i+1)``-th element of `x` starts being valid. 
+- `x::Array{Taylor1{U},N}`: Vector of Taylor polynomials that interpolate the dependent variable as a function of the independent variable.
 """
-@auto_hash_equals struct TaylorInterpolant{T,U,N}
+@auto_hash_equals struct TaylorInterpolant{T, U, N}
     t0::T
-    t::AbstractVector{T}
-    x::AbstractArray{Taylor1{U},N}
+    t::Vector{T}
+    x::Array{Taylor1{U}, N}
```

Comment:
The `TaylorInterpolant` produced by`PlanetaryEphemeris` contains a vector and a matrix with dimensions greater than 100, so I think is not an ideal situation to use e.g. `SVector`.  In this context, I think it is a bigger gain to use a concrete type instead of a abstract type.

Reply:
Maybe `SVector` is not a good example, but think of views or slices, which are not `Vector`s but some subtype of `AbstractVector`...

Reply:
I've changed both fields of `TaylorInterpolant` to their corresponding abstract types.